econml/tests/test_drlearner.py

```diff
@@ -0,0 +1,558 @@
+        dml = LinearDRLearner(model_regression=LinearRegression(),
+                              model_propensity=LogisticRegression(C=1000, solver='lbfgs', multi_class='auto'),
+                              n_splits=KFold(n_splits=3))
+        dml.fit(np.array([1, 2, 3, 1, 2, 3]), np.array([1, 2, 3, 1, 2, 3]), np.ones((6, 1)))
+        dml.score(np.array([1, 2, 3, 1, 2, 3]), np.array([1, 2, 3, 1, 2, 3]), np.ones((6, 1)))
+
+        # test that we can fit with a train/test iterable
+        dml = LinearDRLearner(model_regression=LinearRegression(),
+                              model_propensity=LogisticRegression(C=1000, solver='lbfgs', multi_class='auto'),
+                              n_splits=[([0, 1, 2], [3, 4, 5])])
+        dml.fit(np.array([1, 2, 3, 1, 2, 3]), np.array([1, 2, 3, 1, 2, 3]), np.ones((6, 1)))
+        dml.score(np.array([1, 2, 3, 1, 2, 3]), np.array([1, 2, 3, 1, 2, 3]), np.ones((6, 1)))
+
+    def test_can_use_statsmodel_inference(self):
+        """Test that we can use statsmodels to generate confidence intervals"""
+        dml = LinearDRLearner(model_regression=LinearRegression(),
+                              model_propensity=LogisticRegression(C=1000, solver='lbfgs', multi_class='auto'))
+        dml.fit(np.array([2, 3, 1, 3, 2, 1, 1, 1]), np.array(
+            [3, 2, 1, 2, 3, 1, 1, 1]), np.ones((8, 1)), inference='statsmodels')
+        interval = dml.effect_interval(np.ones((9, 1)),
+                                       T0=np.array([1, 1, 1, 1, 1, 1, 1, 1, 1]),
+                                       T1=np.array([2, 2, 3, 2, 2, 3, 2, 2, 3]),
+                                       alpha=0.05)
+        point = dml.effect(np.ones((9, 1)),
+                           T0=np.array([1, 1, 1, 1, 1, 1, 1, 1, 1]),
+                           T1=np.array([2, 2, 3, 2, 2, 3, 2, 2, 3]))
+        assert len(interval) == 2
+        lo, hi = interval
+        assert lo.shape == hi.shape == point.shape
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+        interval = dml.const_marginal_effect_interval(np.ones((9, 1)), alpha=0.05)
+        point = dml.const_marginal_effect(np.ones((9, 1)))
+        assert len(interval) == 2
+        lo, hi = interval
+        assert lo.shape == hi.shape == point.shape
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+        interval = dml.coef__interval(T=2, alpha=0.05)
+        point = dml.coef_(T=2)
+        assert len(interval) == 2
+        lo, hi = interval
+        assert lo.shape == hi.shape == point.shape
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+    def test_drlearner_all_attributes(self):

```

Comment:
Is there some overlap with the CATE API test?  Could the two tests be consolidated?

Reply:
At some point. I don't know what the CATE API test is exactly testing. I suspect we should have this here too. Double testing is better than no testing and at some point we can clean up. But there might be intricacies with each module and some combinations not caught by general CATE API tests. We should have though general CATE API tests either way to have a global minimal guarantee across modules